### PR TITLE
OWLS-95240 - Reconcile running servers after the domain configuration change and fix for NPE

### PR DESCRIPTION
**Pull request by ankedia**
Changes to reconcile the running servers after a running domain is updated to delete a cluster and/or an independent managed server from the domain topology.. This change removes the server pods for the clustered servers and the independent managed servers which are not part of the domain topology. It also checks if the server config is null before invoking the health check to avoid the NullPointerException.

Integration test run - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7681/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7685/console

### CODE REVIEW

**Review by robertpatrick — APPROVED** [review]
Does this need to be backported to release/3.3?

File: operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import java.util.stream.Collectors;
```
Comment (ddsharpe) [inline operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java:15]: Update all changed files to have copyright date changed to 2022.

**Review by ddsharpe — CHANGES_REQUESTED** [review]

File: operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java (hunk 1 above)
Comment (ankedia) [reply operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java:15]: Changed in ec20ac8.

**Review by jshum2479 — APPROVED** [review]

**Comment by ankedia** [comment]
The latest integration test run has 3 failures - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7712/ . 
I ran the ItElasticLogging and ItElasticLoggingFluentd individually and they passed. The ItStickySession#testSameSessionStickinessUsingClusterService seem to be a known failure. I think the PR is ready to merge.

**Review by robertpatrick — APPROVED** [review]

**Pull request merged**